[Yirui] iP

Zhouyirui opened the pull request:
### Duke

> The one for all Task Manager for everyone

You no longer need to write down or memorize your tasks, with Duke, you can:

- Learn to add, delete tasks with ease
- Access your task manger anywhere
- Super fast to  #learn!!

All you need to do is:

1. Download from[ here](www.youtube.com)
2. Launch the application
3. Start adding tasks!

😉Good news! it is **FREE!**

Features coming soon

- [x] Reminders
- [x] Creating accounts

If you like programming, you can. Here's the `main` method:
```ruby
public static void main(String[] args) {
    new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Zhouyirui committed:
Level-1 of Duke chat bot

Zhouyirui committed:
Level-2 of Duke bot

Zhouyirui committed:
Level-3 of Duke bot

Zhouyirui committed:
Level-4 added Deadline, Event and Todo

Zhouyirui committed:
A-TextUiTesting

Zhouyirui committed:
Level-5 exception handling for missing time. Not yet done for missing task name

Zhouyirui committed:
Delete implemented.

Zhouyirui committed:
some debugging and refactoring

Zhouyirui committed:
Able to read and write from txt file.

Zhouyirui committed:
change from testbranchj

Zhouyirui committed:
Event and Deadline now have a LocalDate

Zhouyirui committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Zhouyirui committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Zhouyirui committed:
Create directory if not found. debugging

Zhouyirui committed:
Add Parser, UI, Storage

Zhouyirui committed:
Add Parser.

Zhouyirui committed:
Organise all classes into duke Package

Zhouyirui committed:
Add JUnit testing for Todo and Deadline

Zhouyirui committed:
Add JavaDoc for Deadline, Event, Storage

Zhouyirui committed:
Follow coding standard

Zhouyirui committed:
no message

sunjc826 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+        LocalDateTime byTime = LocalDateTime.parse(detailD[1], formatter1);
+        Deadline newD = new Deadline(detailD[0], byTime, false);
+        return newD;
+
+    }
```
```suggestion
    }

```

sunjc826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,107 @@
+
+public class Duke {
+    private TaskList tasks;
+//    private String filePath = "data/tasks.txt";
+//    private String directoryName = "data";
```
nit: remove commented out code
I do however think it is quite normal to leave commented code in an in-development app.

sunjc826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,107 @@
+
+                switch (parsedCommand[0]) {
+                case "done":
+                    //Marks tasks as done
```
```suggestion
// Marks tasks as done
```
While this may not be specifically stated in the coding standards, I believe comments look nicer with a space after the `//`.
Likewise for the other inline comments.

sunjc826 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,7 @@
+
+public class DukeException extends Exception {
+    public DukeException() {
+        super("Sorry, duke.Duke cannot understand your command");
```
I think more varied exceptions can be considered. Also, a suggestion: `duke.Duke` could be replaced with just `Duke`.

sunjc826 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+    public String getSavedAs() {
+        return (super.getSavedAs() + "|" + this.at);
+    }
+}
```
In general, files are recommended to have a newline character at the end.

```suggestion
}

```

sunjc826 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+    public int getNumTask() {
+        return numTask;
+    }
+//    private void writeToFile(String text) throws IOException {
+//
+//        fw.write(text);
+//        fw.close();
+//    }
```
```suggestion
```
I believe this responsibility is handled by your `Storage` class, hence this may no longer be necessary.

sunjc826 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,12 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+//import org.junit.jupiter.api.Test;
```
```suggestion
```

sunjc826 reviewed (commented):
On the whole, I like the organization of the code and separation of responsibilities. It is quite clean. My only nitpicks are certain commented out pieces of code, some of which do seem unnecessary and can be deleted safely.

For the future, I also suggest putting a bit more documentation as from what I observe, documentation is only restricted to the class-level and not the methods.

Zhouyirui committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	data/duke.txt

Zhouyirui committed:
Debugging

Zhouyirui committed:
Add testing to Deadline

Zhouyirui committed:
Add JavaDoc comments to Duke, Event, Parser, Storage and TaskList.

Zhouyirui committed:
Tweak code to comply with coding standard

Zhouyirui committed:
Merge branch 'branch-A-CodingStandard'

Zhouyirui committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Zhouyirui committed:
Add Gradle and GUI for Duke

Zhouyirui committed:
Add find

Zhouyirui committed:
Merge branch 'branch-Level-9'

Zhouyirui committed:
Add assertions

Previously, marking a task as done and deleting a task used index.
Index was assumed to be greater than 0.

Assertion was added to document this assumption

Zhouyirui committed:
Merge pull request #2 from Zhouyirui/branch-A-Assertions

Add assertions

Zhouyirui committed:
Set up FXML

Zhouyirui committed:
Improve Code Quality

Methods were very long and difficult to read

Refactored code in Duke. Made long methods shorter.

Zhouyirui committed:
Improve Code Quality

Deleted unnecessary code.

Zhouyirui committed:
Add recurring task

Task that occur weekly can be added.
Only date allowed, no time.

Zhouyirui committed:
Add User Guide

New readers may not know that commands to use with Duke.

In the User Guide, a detailed explanation is given on the types of
commands, how to use the commands, and how Duke will respond to the
commands.

This allows the user to be familiar with using Duke

Zhouyirui committed:
Merge branch 'branch-BCD-Extension'

Zhouyirui committed:
Set theme jekyll-theme-midnight

Zhouyirui committed:
Debug 'bye' command not saving tasks
Create shadowJar file

Zhouyirui committed:
Improve GUI background
Make profile image circular

Zhouyirui committed:
Merge branch 'branch-Level-10'
